flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/utils/WebFrontendBootstrap.java

```diff
@@ -110,10 +114,21 @@ protected void initChannel(SocketChannel ch) {
 		this.serverChannel = ch.sync().channel();
 
 		InetSocketAddress bindAddress = (InetSocketAddress) serverChannel.localAddress();
-		String address = bindAddress.getAddress().getHostAddress();
+
+		InetAddress inetAddress = bindAddress.getAddress();
+		final String address;
+
+		if (inetAddress.isAnyLocalAddress()) {
```

Comment:
What is this case distinction for?

Reply:
This is necessary if the server binds to `0.0.0.0`. Then we have to return the configured address in order to connect to the server.